Metrics announcement (#146)
cloud-gov/site: content/news/articles/2025-06-09-metrics-opensearch.md

---
layout: layouts/post
tags: news

> title: New Metrics for OpenSearch/Elasticsearch Domain Monitoring and S3 Bucket Size

Reviewer: ```suggestion
title: Now available: Metrics for OpenSearch/Elasticsearch domains and S3 buckets
```

> date: 2025-05-08

Reviewer: ```suggestion
date: 2025-06-10
```

> excerpt: "Track detailed performance metrics for your Cloud.gov domains and S3"

Reviewer: ```suggestion
excerpt: "Use detailed performance metrics to monitor your Cloud.gov OpenSearch/Elasticsearch domains and S3 buckets"
```

---

> Cloud.gov has introduced comprehensive Elasticsearch/OpenSearch domain metrics to help teams monitor system performance and resource utilization for brokered domains. As part of this update, Cloud.gov has also implemented a new metric that allows teams to easily track the size of their S3 buckets.

Reviewer: ```suggestion
The Cloud.gov team is excited to announce some new additions to the [logging system][logs] that will allow customers to monitor their services more effectively:

- Comprehensive Elasticsearch/OpenSearch domain metrics to help teams monitor system performance and resource utilization for brokered domains
- A new metric that allows teams to easily track the size of their S3 buckets
```

This update allows customers to access metrics and satisfy [M-21-31 guidelines][m-21-31] for log retention. Metrics are now available in the [Cloud.gov logging system][logs].

## Available S3 Metrics

### Storage and Capacity Metrics

> - **Bucket Size in Bytes**

Reviewer: ```suggestion
- **Bucket Size in Bytes**, metric name: `XXX`
```

Can we add the metric name to each bullet, which will make it easier for people to search them?

## Available Domain Metrics

### System Resource Metrics

#### CPU Utilization
- Overall domain CPU usage
- Master node CPU usage
- Helps identify processing load and potential performance bottlenecks

#### Memory Pressure

> - JVM Memory Pressure
> - Old Generation JVM Memory Pressure
> - Master Node Memory Pressure
> - Monitors memory consumption and potential memory-related issues

Reviewer: ```suggestion
These metrics monitor memory consumption and other potential memory-related issues.

- JVM Memory Pressure
- Old Generation JVM Memory Pressure
- Master Node Memory Pressure
```

### Storage and Capacity Metrics

#### Free Storage Space

> - Available storage in gigabytes
> - Helps track storage capacity and plan for scaling

Reviewer: ```suggestion
This metric helps track storage capacity and plan for scaling.

- Available storage in gigabytes
```

### Threadpool Performance

#### Write Queue Metrics
- Write Queue Count
- Write Queue Rejected Requests

#### Search Queue Metrics
- Search Queue Count
- Search Queue Rejected Requests

> - Monitors background processing and potential system overload

Reviewer: ```suggestion
These metrics monitor background processing and potential system overload:

- Search Queue Count
- Search Queue Rejected Requests
```

{% image "_img/content/add-metrics-filter.png" "Screenshot of OpenSearch Dashboards interface showing the addition of a filter for the @type field with a value of metrics" %}

4. Apply additional filters on the metrics event fields as desired. For example, to filter for domain FreeStorageSpace, add a filter of `metric.name: FreeStorageSpace`:

{% image "_img/content/filter-metric-domain-freestoragespace.png" "Screenshot of OpenSearch Dashboards interface showing a filtered search for the metric FreeStorageSpace" %}

## Metrics Fields
The fields available on metrics records are:
- `@cf.service_offering` - Type of service being monitored
- `@cf.service_plan` - Specific service plan details
- `metric.average` - Numeric value of the metric
- `metric.name` - Name of the specific metric
- `metric.unit` - Unit of measurement

## How Metrics Are Ingested into the Logging System
An automated job runs every 10 minutes to pull metric events from the platform and ingest them into the logging system. AWS limits most metrics to a 2-minute delay. Thus, **there could be up to a 12-minute delay** before metric logs appear in the logging system.

## Retention

> Metric events are retained in the logging system for 12 months and in offline storage for an additional 18 months.

Reviewer: ```suggestion
Metric events are retained in the logging system for 12 months and in offline storage for an additional 18 months in accordance with [M-21-31 guidelines][m-21-31].
```
